Modules/ThirdParty/OssimPlugins/src/ossim/ossimSentinel1Model.cpp

```diff
@@ -505,21 +509,44 @@ namespace ossimplugins
 
    bool ossimSentinel1Model::checkDirectory(const ossimFilename& file, const char* d, const char* ext) const
    {
-      //check dir is valid first
-      ossimDirectory dir(file.path() + '/' + d + '/');
-      std::vector<ossimFilename> result;
-      dir.findAllFilesThatMatch(result, ext);
-      if ( result.empty() )
-      {
-         if (traceExec())
-         {
+    std::stringstream strm;
+    std::vector<ossimFilename> result;
+
+	#if defined(_WIN32)
+		const char pathsep = '\\';
+	#else
+		const char pathsep = '/';
+	#endif
+
+	#if defined(_MSC_VER)
+	strm << file.path() << "\\" << d << pathsep << "*" << ext;
+	WIN32_FIND_DATA search_data;
+	memset(&search_data, 0, sizeof(WIN32_FIND_DATA));
+	HANDLE handle = FindFirstFile(strm.str().c_str(), &search_data);
+	while(handle != INVALID_HANDLE_VALUE)  {
+	  result.push_back(std::string( search_data.cFileName ) );
+      if(FindNextFile(handle, &search_data) == FALSE)
+        break;
+	}
+	//Close the handle
+	FindClose(handle);
```

Comment:
There is a possible resource leak here as there is no guarantee that `FClose` will be called in case the string construction or `push_back` fail.